app/controllers/orders_controller.rb

```diff
@@ -1,35 +1,47 @@
 class OrdersController <ApplicationController
 
   def new
-
   end
 
   def show
     @order = Order.find(params[:id])
   end
 
   def create
-    order = Order.create(order_params)
-    if order.save
-      cart.items.each do |item,quantity|
-        order.item_orders.create({
-          item: item,
-          quantity: quantity,
-          price: item.price
-          })
-      end
-      session.delete(:cart)
-      redirect_to "/orders/#{order.id}"
-    else
-      flash[:notice] = "Please complete address form to create an order."
-      render :new
-    end
+    order = Order.new(order_params)
+    can_create_order? ? create_new(order) : retry_order_creation
```

Comment:
@ajtran303 can you explain this syntax or link to documentation about it?

Reply:
I am using a ternary operator to flatten an if/else statement.

```ruby
condition ? true : false

if_condition_is_true ? then_do_this : else_do_that
```

Reply:
[Here is a guide](https://www.rubyguides.com/2019/10/ruby-ternary-operator/)

Reply:
Thanks, looks like a good use case for it here.